h/api/search/query.py

```diff
@@ -68,25 +88,25 @@ def build(request_params, userid=None):
 
     for key, value in request_params.items():
         matches.append({"match": {key: value}})
-    matches = matches or [{"match_all": {}}]
 
-    query["query"] = {"bool": {"must": matches}}
+    # Add a filter for "not in public site areas" considerations
+    filters.append(nipsa.nipsa_filter(userid=userid))
 
-    query["query"] = {
-        "filtered": {
-            "filter": nipsa.nipsa_filter(userid=userid),
-            "query": query["query"]
-        }
-    }
+    # And now we simplify the query. If there are no added matches, then we can
```

Comment:
This isn't really intended to be optimization for ElasticSearch. It was supposed to be optimization for humans trying to read the generated queries while debugging :)

But certainly we can do something more like what you suggested in your second comment.